src/model/paths_types.ml

```diff
@@ -592,10 +597,16 @@ module rec Reference : sig
     | `TChildPage
     | `TChildModule ]
 
+  type page_path =
+    [ `Root of string * tag_page_path
+    | `Slash of page_path * string (* {!page_path/identifier} *) ]
```

Comment:
I don't think `identifier` should be a string: We agreed that it should be parsed as a page reference, so we can write:

```
{!page_path/page-pagename.section-sectionname}
```

Reply:
Ah in fact maybe it should be a string, but the identifier part should be present somewhere else. (see next comment!)

Reply:
Indeed, there's a problem here. I think this is related to the lack of a `Module_path` constructor. The prefix can only appear on the last component of the path and the intended representation is more like this:
```
Label (Page_path (Slash (Root ("page_path", relative), "pagename")), "sectionname")
```

The result for `{!page_path/page-pagename}` is as intended but there's two problems:
- This reference with a label do not parse.
- No representation for `path/module-Bar`.

I'll look into it.